src/test/java/app/bpartners/api/unit/service/CustomerServiceTest.java

```diff
@@ -0,0 +1,133 @@
+import app.bpartners.api.service.utils.GeoUtils;
+import java.util.*;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class CustomerServiceTest {
+  CustomerRepository repositoryMock;
+  BanApi banApiMock;
+  CustomerRestMapper restMapperMock;
+  EventProducer eventProducerMock;
+  EventConf eventConfMock;
+  SesConf sesConfMock;
+  CustomerService subject;
+
+  @BeforeEach
+  void setUp() {
+    repositoryMock = mock(CustomerRepository.class);
+    banApiMock = mock(BanApi.class);
+    subject =
+        new CustomerService(
+            repositoryMock,
+            restMapperMock,
+            eventProducerMock,
+            eventConfMock,
+            sesConfMock,
+            banApiMock);
+  }
+
+  Customer.CustomerBuilder customerBuilder() {
+    return Customer.builder().id(JOE_DOE_ID);
+  }
+
+  @Test
+  void testUpdateCustomersLocationAddressTooShort() {
+    var customer = customerBuilder().address("12").build();
+    var customers = mock(List.class);
+
+    when(customers.get(anyInt())).thenReturn(customer);
+    when(repositoryMock.findWhereLatitudeOrLongitudeIsNull()).thenReturn(customers);
+
+    subject.updateCustomersLocation();
+    verify(repositoryMock, never()).save(any(Customer.class));
+  }
+
+  @Test
+  void testUpdateCustomersLocationAddressTooLong() {
+    var customer = customerBuilder().address("A".repeat(201)).build();
+    var customers = mock(List.class);
+
+    when(customers.get(anyInt())).thenReturn(customer);
+    when(repositoryMock.findWhereLatitudeOrLongitudeIsNull()).thenReturn(customers);
+
+    subject.updateCustomersLocation();
+    verify(repositoryMock, never()).save(any(Customer.class));
+  }
+
+  @Test
+  void testUpdateCustomersLocationAddressNotFound() {
+    var customer = customerBuilder().address("123 Street").build();
+    var customers = mock(List.class);
+
+    when(customers.get(anyInt())).thenReturn(customer);
+    when(repositoryMock.findWhereLatitudeOrLongitudeIsNull()).thenReturn(customers);
+    when(banApiMock.search("123 Street")).thenReturn(null);
+
+    subject.updateCustomersLocation();
+    verify(repositoryMock, never()).save(any(Customer.class));
+  }
+
+  @Test
+  void testUpdateCustomersLocationExceptionHandling() {
+    var customer = customerBuilder().address("123 Street").build();
+    var customers = mock(List.class);
+
+    when(customers.get(anyInt())).thenReturn(customer);
+    when(repositoryMock.findWhereLatitudeOrLongitudeIsNull()).thenReturn(customers);
+    when(banApiMock.search(any())).thenThrow(new BadRequestException("Bad Request"));
+
+    subject.updateCustomersLocation();
+    verify(repositoryMock, never()).save(any(Customer.class));
+  }
+
+  @Test
+  void testUpdateCustomersLocationSuccessfulUpdate() {
+    var geoPosition = mock(GeoPosition.class);
+    var coordinates = mock(GeoUtils.Coordinate.class);
+    var customer = mock(Customer.class);
+
+    when(customer.getId()).thenReturn("1");
```

Comment:
No. Anti-pattern. Why mocking when you can create the object?

Reply:
This variable needs to be mocked, as verify is used only with mocks.